app/routes/claim/file-upload.js

```diff
@@ -20,52 +22,69 @@ module.exports = function (router) {
-        claimId = req.params.claimId
-      } else {
-        claimId = null
-      }
-      DirectoryCheck(`${req.params.referenceId}-${req.query.eligibilityId}`, claimId, req.query.claimExpenseId, req.params.documentType)
       return res.render('claim/file-upload', {
         claimType: req.params.claimType,
         document: req.params.documentType,
         fileUploadGuidingText: DocumentTypeEnum,
         URL: req.url
       })
-    } else {
-      throw new Error('Not a valid document type')
     }
+
+    throw new Error('Not a valid document type')
   })
 
   router.post('/claim/file-upload/:referenceId/:claimId/:documentType', function (req, res, next) {
     authorisation.hasRoles(req, allowedRoles)
 
-    Upload(req, res, function (error) {
+    Upload(req, res, async function (error) {
       try {
         // If there was no file attached, we still need to check the CSRF token
         if (!req.file) {
           csrfProtection(req, res, function (error) {
             if (error) { throw error }
           })
         }
+
         if (error) {
           if (error.message === 'File too large') {
             throw new ValidationError({ upload: [ERROR_MESSAGES.getUploadTooLarge] })
           } else {
             throw error
           }
-        } else {
-          if (Object.prototype.hasOwnProperty.call(DocumentTypeEnum, req.params.documentType)) {
-            const fileUpload = new FileUpload(req.file, req.error, req.query.claimDocumentId, req.user.email)
-            ClaimDocumentUpdate(req.params.referenceId, fileUpload).then(function () {
-              res.redirect(`/claim/${req.params.claimId}`)
-            }).catch(function (error) {
-              next(error)
-            })
+        }
+
+        const originalUploadPath = getFileUploadPath(req)
+
+        if (Object.prototype.hasOwnProperty.call(DocumentTypeEnum, req.params.documentType)) {
+          const filenamePrefix = getFilenamePrefix(req.params, req.query)
+          const filename = getUploadFilename(req)
+          const targetFileName = `${filenamePrefix}/${filename}`
+
+          if (req.file) {
+            req.file.destination = ''
+            req.file.path = targetFileName
           } else {
-            throw new Error('Not a valid document type')
+            req.file = {
+              destination: '',
+              path: targetFileName
+            }
+          }
+
+          const fileUpload = new FileUpload(req.file, req.error, req.query.claimDocumentId, req.user.email)
+
+          try {
+            await aws.upload(targetFileName, originalUploadPath)
+          } catch (error) {
+            next(error)
           }
+
+          ClaimDocumentUpdate(req.params.referenceId, fileUpload).then(function () {
+            res.redirect(`/claim/${req.params.claimId}`)
+          }).catch(function (error) {
+            next(error)
+          })
+        } else {
```

Comment:
could possibly invert your `if` and throw this early as a bit of an assertion check..

Reply:
Good call, hadn't spotted 👍